src/Mesh/Interpolation.jl

```diff
@@ -1,36 +1,58 @@
 module Interpolation
+using DocStringExtensions
 using CLIMA
 using MPI
 import GaussQuadrature
 using CLIMA.Mesh.Topologies
 using CLIMA.Mesh.Grids
 using CLIMA.Mesh.Geometry
 using CLIMA.Mesh.Elements
-using CLIMA.Writers
 using LinearAlgebra
 using StaticArrays
+import KernelAbstractions: CPU, CUDA
 #-------------------
 using CUDAnative
-using KernelAbstractions: CPU, CUDA
+using CuArrays
 
 export InterpolationBrick,
-    write_interpolated_data,
+    accumulate_interpolated_data!,
     InterpolationCubedSphere,
     interpolate_local!,
+    project_cubed_sphere!,
     InterpolationTopology
 
 abstract type InterpolationTopology end
 #--------------------------------------------------------
+#--------------------------------------------------------
 """
-    InterpolationBrick(grid::DiscontinuousSpectralElementGrid{FT}, xbnd::Array{FT,2}, xres) where FT <: AbstractFloat
+    InterpolationBrick{
+    FT <: AbstractFloat,
+    T <: Int,
+    FTV <: AbstractVector{FT},
+    FTVD <: AbstractVector{FT},
+    TVD <: AbstractVector{T},
+    FTA2 <: Array{FT, 2},
+    UI8AD <: AbstractArray{UInt8, 2},
+    UI16VD <: AbstractVector{UInt16},
+    I32V <: AbstractVector{Int32},
+    } <: InterpolationTopology
 
-This interpolation structure and the corresponding functions works for a brick, where stretching/compression happens only along the x, y & z axis.
+This interpolation data structure and the corresponding functions works for a brick, where stretching/compression happens only along the x, y & z axis.
 Here x1 = X1(ξ1); x2 = X2(ξ2); x3 = X3(ξ3)
 
-# input for the inner constructor
- - `grid` DiscontinousSpectralElementGrid
- - `xbnd` Domain boundaries in x1, x2 and x3 directions
- - `xres` Resolution of the interpolation grid in x1, x2 and x3 directions
+# Fields
+
+$(DocStringExtensions.FIELDS)
+
+# Usage
+    InterpolationBrick(grid::DiscontinuousSpectralElementGrid{FT}, xbnd::Array{FT,2}, xres) where FT <: AbstractFloat
+    This interpolation structure and the corresponding functions works for a brick, where stretching/compression happens only along the x, y & z axis.
+    Here x1 = X1(ξ1); x2 = X2(ξ2); x3 = X3(ξ3)
+    # input for the inner constructor
+     - `grid` DiscontinousSpectralElementGrid
+     - `xbnd` Domain boundaries in x1, x2 and x3 directions
+     - `xres` Resolution of the interpolation grid in x1, x2 and x3 directions
```

Comment:
Only the code should be indented, regular text should be left-aligned (and inline code quoted with backticks):

```suggestion
# Usage

    InterpolationBrick(grid::DiscontinuousSpectralElementGrid{FT}, xbnd::Array{FT,2}, xres) where FT <: AbstractFloat
    
This interpolation structure and the corresponding functions works for a brick, where stretching/compression happens only along the x, y & z axis. Here `x1 = X1(ξ1); x2 = X2(ξ2); x3 = X3(ξ3)`

Arguments for the inner constructor
 - `grid`: `DiscontinousSpectralElementGrid`
 - `xbnd` Domain boundaries in `x1`, `x2` and `x3` directions
 - `xres` Resolution of the interpolation grid in `x1`, `x2` and `x3` directions
```